pages/[resourceType]/create.tsx

```diff
@@ -49,6 +48,72 @@ const CreateResourcePage = () => {
       </Center>
     </div>
   );
+
+  //handles what will happen when the submit button is clicked.
+  function editorSubmitHandler() {
+    let notifProps = {
+      message: <div>Problem connecting to server</div>,
+      color: "red",
+      icon: <X size={18} />,
+      autoClose: false,
+    };
+
+    fetch(`${process.env.NEXT_PUBLIC_DEQM_SERVER}/${resourceType}`, {
+      method: "POST",
+      body: codeEditorContents,
+      headers: {
+        "Content-Type": "application/json+fhir",
+      },
+    })
+      .then((response) => {
+        let newID;
+
+        if (response.status === 201 || response.status === 200) {
+          newID = resourceType
+            ? response.headers.get("Location")?.substring(7 + resourceType.length)
+            : "";
+
+          notifProps = {
+            ...notifProps,
+            //new ID rendered as a Link that navigates to page displaying the resource's body
+            message: (
+              <>
+                <Text>Resource successfully created:&nbsp;</Text>
+                <Link href={`/${resourceType}/${newID}`} key={`${resourceType}/${newID}`} passHref>
+                  <Text component="a" color="cyan">
+                    {resourceType}/{newID}
+                  </Text>
+                </Link>
+              </>
+            ),
+            color: "green",
+            icon: <Check size={18} />,
+          };
+          //redirects user to the resourceType home page
+          router.push({ pathname: `/${resourceType}` });
+        } else {
```

Comment:
In this else case, we know that the server must have responded, but with an error code of some sort. `deqm-test-server` handles errors by responding with a [FHIR OperationOutcome Resource](http://www.hl7.org/fhir/operationoutcome.html), which is standard for all FHIR servers.

A more detailed error message lives in the response body, e.g.

```
{
	"resourceType": "OperationOutcome",
	"issue": [
		{
			"severity": "error",
			"code": "BadRequest",
			"details": {
				"text": "resourceType: undefined is not a supported resourceType"
			}
		}
	]
}
```

I wonder if there's a way where, in this else case, we return another promise (`return response.json()`) and chain on another `.then`. This `.then` will contain the OperationOutcome error body. So if that's defined, then we can pull off the error details from `.issue[0].details.text`?